resip/stack/TransportSelector.cxx

```diff
@@ -1638,37 +1633,47 @@ TransportSelector::findTransportBySource(Tuple& search, const SipMessage* msg) c
 }
 
 Transport*
-TransportSelector::findTlsTransport(const Data& domainname, TransportType type, IpVersion version) const
+TransportSelector::findTlsTransport(const Data& domainname,
+                                    const Tuple& search) const
 {
-   resip_assert(isSecure(type));
-   DebugLog(<< "Searching for " << toData(type) << " transport for domain='"
-                  << domainname << "'" << " have " << mTlsTransports.size());
+   resip_assert(isSecure(search.getType()));
+
+   // Basically, the 'search' Tuple has a member called mTargetDomain.
+   // However, it is not used due to the current TransportSelector implementation logic.
 
    if (domainname == Data::Empty)
    {
-      for(TlsTransportMap::const_iterator i=mTlsTransports.begin(); i != mTlsTransports.end();++i)
+      DebugLog(<< "Searching for " << toData(search.getType()) << " transport without domain."
+               << " Secure transports list size = " << mTlsTransports.size());
+
+      for(const auto& tlsTransport : mTlsTransports)
       {
-         if(i->first.mTuple.getType() == type && i->first.mTuple.ipVersion() == version)
+         const TlsTransportKey &key = tlsTransport.first;
+
+         if (key.mTuple == search)  // Tuple does not compare domain names by default.
          {
-            DebugLog(<<"Found a default transport.");
-            return i->second;
+            DebugLog(<< "findTlsTransport (exact match) => " << *(tlsTransport.second));
+            return tlsTransport.second;
          }
```

Comment:
If we fail to match with Tuple ==, should we fallback to the old code matching and just try to find match on type and ipversion?  We could keep track of the "best" match as we iterate through the list.  Just worried about making sure we are backwards compatible here.